src/iam/certhandler/certhandler.cpp

```diff
@@ -140,21 +138,45 @@ Error CertHandler::GetCertificate(
         return AOS_ERROR_WRAP(err);
     }
 
-    LOG_DBG() << "Get certificate: type = " << certType << ", serial = " << serialInHex;
+    LOG_DBG() << "Subscribe certificate receiver: type = " << certType << ", serial = " << serialInHex;
+
+    err = mCertReceiverSubscriptions.EmplaceBack(certType, issuer, serial, &certReceiver);
+    if (!err.IsNone()) {
+        return AOS_ERROR_WRAP(err);
+    }
+
+    auto*    module = FindModule(certType);
+    CertInfo certInfo;
 
-    auto* module = FindModule(certType);
     if (module == nullptr) {
         return AOS_ERROR_WRAP(ErrorEnum::eNotFound);
     }
 
-    err = module->GetCertificate(issuer, serial, resCert);
+    err = module->GetCertificate(issuer, serial, certInfo);
     if (!err.IsNone()) {
         return AOS_ERROR_WRAP(err);
     }
 
+    certReceiver.UpdateCert(certInfo);
+
     return ErrorEnum::eNone;
 }
 
+Error CertHandler::UnsubscribeCertReceiver(CertReceiverItf& certReceiver)
+{
+    LockGuard lock(mMutex);
```

Comment:
LockGuard lock {mMutex};

Reply:
updated in a separate pr for all the files.